txpool/txpool.go

```diff
@@ -659,13 +659,13 @@ func (p *TxPool) validateTx(tx *types.Transaction) error {
 
 			return ErrUnderpriced
 		}
-	} else {
-		// Legacy approach to check if the given tx is not underpriced
-		if tx.GetGasPrice(baseFee).Cmp(big.NewInt(0).SetUint64(p.priceLimit)) < 0 {
-			metrics.IncrCounter([]string{txPoolMetrics, "underpriced_tx"}, 1)
+	}
 
-			return ErrUnderpriced
-		}
+	// Check if the given tx is not underpriced
+	if tx.GetGasPrice(baseFee).Cmp(big.NewInt(0).SetUint64(p.priceLimit)) < 0 {
```

Comment:
why not `new(big.Int).SetUInt64(p.PriceLimit)`?

Reply:
Same question :)